client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts

```diff
@@ -94,13 +95,22 @@ const useInvestigationTable = (): useInvestigationTableOutcome => {
       });
   }, [user.id, classes.errorAlertTitle]);
 
-
-  const onInvestigationRowClick = (epidemiologyNumber: number) => {
+  const onInvestigationRowClick = (epidemiologyNumberVal: number) => {
+    axios.interceptors.request.use(
+      (config) => {
+          config.headers.Authorization = user.token;
+          config.headers.EpidemiologyNumber = epidemiologyNumberVal;
+          config.headers.UserName = 'stub_user'
+          setIsLoading(true);
+          return config;
+      },
+      (error) => Promise.reject(error)
+  );
```

Comment:
Export it to a useEffect in App.tsx that any time that the redux changes then the headers change as well

Reply:
תיקנתי